### PR TITLE
Cluster reload

### PR DESCRIPTION
**Pull request by tylertreat**
Adds support for cluster/routes config reload. Routes authorization behavior is similar to users: if username/password change, routes that are no longer authorized are closed.

@nats-io/core

### CODE REVIEW

**Comment by tylertreat** [comment]
The test is currently failing because when we remove a route from the `routes` list, the route is closed but the remote peer initiates a reconnect which reestablishes the route. Not sure how best to handle this case or what the proper behavior should be. Thoughts @derekcollison @petemiron @ColinSullivan1?

**Comment by petemiron** [comment]
@tylertreat I would expect this feature to only be used after a host has been removed from a cluster -- eg. after shutdown of the server. Would it make sense to shutdown the server, then reload the cluster and test to make sure the server stops trying to reconnect to that removed route?

**Comment by tylertreat** [comment]
@petemiron I agree that is most likely the common case, though it seems like it would be useful to allow reconfiguring a cluster without bringing down/up instances, e.g. splitting a single cluster into two clusters by reconfiguring routes. The current behavior would preclude that.

**Comment by petemiron** [comment]
We'd need a way to reload cluster config even if there's no change to
config.

**Comment by tylertreat** [comment]
I'm not sure what you mean.

**Comment by tylertreat** [comment]
Followed up with @petemiron offline. The thought is we might want to add a flag that allows dropping discovered routes for cases where an operator is intentionally partitioning a cluster. However, I don't think this solves the reconnect loop, so this probably needs further discussion. Outside the scope of this PR.

File: util/tls.go [hunk 1]
```diff
@@ -1,37 +1,13 @@
...
-		SessionTicketsDisabled:      c.SessionTicketsDisabled,
-		SessionTicketKey:            c.SessionTicketKey,
-		ClientSessionCache:          c.ClientSessionCache,
-		MinVersion:                  c.MinVersion,
-		MaxVersion:                  c.MaxVersion,
-		CurvePreferences:            c.CurvePreferences,
-		DynamicRecordSizingDisabled: c.DynamicRecordSizingDisabled,
-		Renegotiation:               c.Renegotiation,
-	}
+	return c.Clone()
```
Comment (petemiron) [inline util/tls.go]: that's convenient.

File: server/reload_test.go [hunk 2]
```diff
@@ -1239,3 +1256,427 @@ func TestConfigReloadChangePermissions(t *testing.T) {
 	default:
 	}
 }
+
+// Ensure Reload returns an error when attempting to change cluster address
+// host.
+func TestConfigReloadClusterHostUnsupported(t *testing.T) {
+	dir, err := os.Getwd()
```
Comment (petemiron) [inline server/reload_test.go]: I'd refactor this loading and server startup into a single func.

**Pull request merged**